client/lib/support/support-user/localforage-bypass.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+
+const debug = debugModule( 'calypso:support-user' );
+
+// This module defines a custom localForage driver which bypasses all persistent
+// storage. Any calls to read/write data using localForage instead access a temporary
+// in-memory store which is lost on page reload. This driver is used to sandbox
+// a user's data while support-user is active, ensuring it does not contaminate the
+// original user/vice versa.
+// Copied with modifications from localForage source: localforage/test/dummyStorageDriver.js
+
+const dummyStorage = {};
+
+// Config the localStorage backend, using options set in the config.
+function _initStorage( options ) {
+	const dbInfo = {};
+	if ( options ) {
+		for ( let i in options ) {
+			dbInfo[ i ] = options[ i ];
```

Comment:
Seems these handful of lines be replaced with a single [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)?

```js
const dbInfo = Object.assign( {}, options, { db: {} } );
```